subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyUnresolvedModuleIntegrationTest.groovy

```diff
@@ -0,0 +1,407 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve
+
+import org.gradle.integtests.fixtures.AbstractHttpDependencyResolutionTest
+import org.gradle.test.fixtures.file.TestFile
+import org.gradle.test.fixtures.keystore.TestKeyStore
+import org.gradle.test.fixtures.maven.MavenFileRepository
+import org.gradle.test.fixtures.maven.MavenModule
+import org.gradle.test.fixtures.maven.MavenRepository
+import org.gradle.test.fixtures.server.http.HttpResource
+import org.gradle.test.fixtures.server.http.MavenHttpModule
+import org.gradle.test.fixtures.server.http.MavenHttpRepository
+import spock.lang.Unroll
+
+import static org.gradle.internal.resource.transport.http.JavaSystemPropertiesHttpTimeoutSettings.SOCKET_TIMEOUT_SYSTEM_PROPERTY
+
+class DependencyUnresolvedModuleIntegrationTest extends AbstractHttpDependencyResolutionTest {
+
+    private static final String GROUP_ID = 'group'
+    private static final String VERSION = '1.0'
+    TestFile downloadedLibsDir
+    MavenHttpModule moduleA
+
+    def setup() {
+        moduleA = publishMavenModule(mavenHttpRepo, 'a')
+        downloadedLibsDir = file('build/libs')
+        executer.withArgument("-D${SOCKET_TIMEOUT_SYSTEM_PROPERTY}=1000")
+    }
+
+    void blockingForProtocol(String protocol, HttpResource... resources) {
+        if (protocol == 'http') {
+            resources.each { it.expectGetBlocking() }
+        } else if (protocol == 'https') {
+            // https://issues.apache.org/jira/browse/HTTPCLIENT-1478
+            def keyStore = TestKeyStore.init(temporaryFolder.file('ssl-keystore'))
+            keyStore.enableSslWithServerCert(server)
+            keyStore.configureServerCert(executer)
+            server.expectSslHandshakeBlocking()
+        } else {
+            assert false
+        }
+    }
+
+    @Unroll
+    def "fails single build script dependency resolution if #protocol connection exceeds timeout"() {
+        given:
+        blockingForProtocol(protocol, moduleA.pom)
+        buildFile << """
+            buildscript {
+                ${mavenRepository(mavenHttpRepo)}
+
+                dependencies {
+                    classpath '${mavenModuleCoordinates(moduleA)}'
+                }
+            }
+        """
+
+        when:
+        fails('help')
+
+        then:
+        assertDependencyMetaDataReadTimeout(moduleA)
+
+        where:
```

Comment:
This is missing some comments on the PR that succeeded this one e.g. https://github.com/gradle/gradle/pull/3041#discussion_r141990741. It had been fixed in the PR. I feel like there's still something wrong with what has or has not been merged. Why does this change even show up in this PR? It should already be in `release`.

Reply:
Good catch, I did miss something when merging and I've fixed this. The reason why this test shows up here is this file was renamed from 
 `DependencyResolveTimeoutIntegrationTest` to `DependencyUnresolvedModuleIntegrationTest` in https://github.com/gradle/gradle/commit/65c833548f8fb3d3c5097cd3db8545979aaab43c

Reply:
Maybe I also mixed something when I rebased this PR. `DependencyResolveTimeoutIntegrationTest` is not needed anymore then, because everything is in `DependencyUnresolvedModuleIntegrationTest` now, right? Can you remove `DependencyResolveTimeoutIntegrationTest`?

Reply:
Oh yes, sure.